src/Components/test/testassets/BasicTestApp/FormsTest/NotifyPropertyChangedValidationComponent.cshtml

```diff
@@ -0,0 +1,105 @@
+@using System.ComponentModel
+@using System.ComponentModel.DataAnnotations
+@using System.Runtime.CompilerServices;
+@using Microsoft.AspNetCore.Components.Forms
+
+<p>
+    There's no requirement for models to implement INotifyPropertyChanged, but if they do,
+    you can easily wire that up to the EditContext. Then you have no need to use the built-in
+    Input* components - you can instead bind to regular HTML elements and still get modification
+    notifications. This provides more flexibility in how the UI is rendered, at the cost of
+    more complexity and boilerplate in your model classes.
+</p>
+<p>
+    This example also shows that you don't strictly have to use EditForm. You can manually
+    cascade an EditContext to the components that integrate with it. 
+</p>
+
+<form onsubmit="@HandleSubmit">
+    <p>
+        User name:
+        <input bind="@person.UserName" class="@editContext.FieldClass(() => person.UserName)" />
+    </p>
+    <p>
+        Accept terms:
+        <input type="checkbox" bind="@person.AcceptsTerms" class="@editContext.FieldClass(() => person.AcceptsTerms)" />
+    </p>
+
+    <button type="submit">Submit</button>
+
+    <CascadingValue Value="@editContext" IsFixed="true">
+        <ValidationSummary />
+    </CascadingValue>
+</form>
+
+@submissionStatus
+
+@functions {
+    MyModel person = new MyModel();
+    EditContext editContext;
+    string submissionStatus;
+
+    protected override void OnInit()
+    {
+        editContext = new EditContext(person).AddDataAnnotationsValidation();
+
+        // Wire up INotifyPropertyChanged to the EditContext
+        person.PropertyChanged += (sender, eventArgs) =>
+        {
+            var fieldIdentifier = new FieldIdentifier(sender, eventArgs.PropertyName);
+            editContext.NotifyFieldChanged(fieldIdentifier);
+        };
```

Comment:
I think it's OK to have this be manual for now. If we see people doing it in this release we could consider baking it in.

Reply:
Yes. I shied away from baking it in because (1) it forces another dependency on `System.ComponentModel` and (2) so far we've consistently said about `INotifyPropertyChange`/`INotifyDataErrorInfo`/etc that we're not taking an opinion on them but rather are designing APIs to be convenient to integrate them (and other things) with.

But if this does become a popular option we definitely should consider that.